cmd/integrationArtifactTriggerIntegrationTest.go

```diff
@@ -0,0 +1,156 @@
+	// and use a  &piperhttp.Client{} in a custom system
+	// Example: step checkmarxExecuteScan.go
+
+	// Error situations should be bubbled up until they reach the line below which will then stop execution
+	// through the log.Entry().Fatal() call leading to an os.Exit(1) in the end.
+	err := runIntegrationArtifactTriggerIntegrationTest(&config, telemetryData, utils, httpClient)
+	if err != nil {
+		log.Entry().WithError(err).Fatal("step execution failed")
+	}
+}
+
+func runIntegrationArtifactTriggerIntegrationTest(config *integrationArtifactTriggerIntegrationTestOptions, telemetryData *telemetry.CustomData, utils integrationArtifactTriggerIntegrationTestUtils, httpClient piperhttp.Sender) error {
+	var commonPipelineEnvironment integrationArtifactGetServiceEndpointCommonPipelineEnvironment
+	var serviceEndpointUrl string
+	if len(config.IFlowServiceEndpointURL) > 0 {
+		serviceEndpointUrl = config.IFlowServiceEndpointURL
+	} else {
+		serviceEndpointUrl = commonPipelineEnvironment.custom.iFlowServiceEndpoint
+		if len(serviceEndpointUrl) == 0 {
+			log.SetErrorCategory(log.ErrorConfiguration)
+			return fmt.Errorf("IFlowServiceEndpointURL not set")
+		}
+	}
+	log.Entry().Info("The Service URL : ", serviceEndpointUrl)
+
+	// Here we trigger the iFlow Service Endpoint.
+	IFlowErr := callIFlowURL(config, telemetryData, utils, httpClient, serviceEndpointUrl)
+	if IFlowErr != nil {
+		log.SetErrorCategory(log.ErrorService)
+		return fmt.Errorf("failed to execute iFlow: %w", IFlowErr)
+	}
+
+	return nil
+}
+
+func callIFlowURL(config *integrationArtifactTriggerIntegrationTestOptions, telemetryData *telemetry.CustomData, utils integrationArtifactTriggerIntegrationTestUtils, httpIFlowClient piperhttp.Sender, serviceEndpointUrl string) error {
+
+	var fileBody []byte
+	var httpMethod string
+	var header http.Header
+	if len(config.MessageBodyPath) > 0 {
+		if len(config.ContentType) == 0 {
+			log.SetErrorCategory(log.ErrorConfiguration)
+			return fmt.Errorf("message body file %s given, but no ContentType", config.MessageBodyPath)
+		}
+		exists, err := utils.FileExists(config.MessageBodyPath)
+		if err != nil {
+			log.SetErrorCategory(log.ErrorUndefined)
+			// Always wrap non-descriptive errors to enrich them with context for when they appear in the log:
+			return fmt.Errorf("failed to check message body file %s: %w", config.MessageBodyPath, err)
+		}
+		if !exists {
+			log.SetErrorCategory(log.ErrorConfiguration)
+			return fmt.Errorf("message body file %s configured, but not found", config.MessageBodyPath)
+		}
+
+		var fileErr error
+		fileBody, fileErr = ioutil.ReadFile(config.MessageBodyPath)
+		if fileErr != nil {
+			log.SetErrorCategory(log.ErrorUndefined)
+			return fmt.Errorf("failed to read file %s: %w", config.MessageBodyPath, fileErr)
+		}
+		httpMethod = "POST"
+		header = make(http.Header)
+		header.Add("Content-Type", config.ContentType)
+	} else {
+		httpMethod = "GET"
```

Comment:
This looks strange. If there is no message body we trigger the integration tests with a GET request?